File: solid/Daniel Cueto - Luis Enrique Gamboa/src/EjercicioAssure/Empresa.java
package EjercicioAssure;
import java.util.ArrayList;
import java.util.List;

public class Empresa {
    private List<Empleado> empleados;

    public Empresa() {
        this.empleados = new ArrayList<>();
    }
    
    public void agregarEmpleado(Empleado empleado) {
        this.empleados.add(empleado);
    }
    
    public double calcularPlanilla() {
        double total = 0;
        for (Empleado e : empleados) {
            total += e.calcularSalario();
        }
        return total;
    }
    
    public void mostarEmpleados() {
        for (Empleado e : empleados) {
            System.out.println(e);
        }
    }  
}

File: solid/max_miguel/typescript/Empresa.ts
import Empleado from "./Empleado";

export default class Empresa {
  private empleados: Empleado[];

  constructor() {
    this.empleados = [];
  }

  public agregarEmpleado(empleado: Empleado): void {
    this.empleados.push(empleado);
  }

  public calcularPlanillaTotal(): number {
    let total: number = 0;
    for (const empleado of this.empleados) {
      total += empleado.calcularSueldo();
    }
    return total;
  }
}
